corehq/apps/data_interfaces/forms.py

```diff
@@ -94,3 +99,179 @@ def __init__(self, *args, **kwargs):
+        required=False,
+    )
+    update_case = forms.BooleanField(
+        label=ugettext_lazy("Set a case property"),
+        required=False,
+    )
+    update_property_name = TrimmedCharField(
+        label=ugettext_lazy("Property"),
+        required=False,
+    )
+    update_property_value = TrimmedCharField(
+        label=ugettext_lazy("Value"),
+        required=False,
+    )
+
+    @property
+    def current_values(self):
+        values = {}
+        for field_name in self.fields.keys():
+            value = self[field_name].value()
+            if field_name == 'conditions':
+                value = value or '[]'
+                value = json.loads(value)
+            values[field_name] = value
+        return values
+
+    def __init__(self, *args, **kwargs):
+        super(AddAutomaticCaseUpdateRuleForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = 'form form-horizontal'
+        self.helper.label_class = 'col-sm-1 col-md-1'
+        self.helper.field_class = 'col-sm-3 col-md-2'
+        self.helper.form_method = 'POST'
+        self.helper.form_action = '#'
+        self.helper.layout = Layout(
+            Fieldset(
+               _("Basic Information"),
+                Field(
+                    'name',
+                    **{'ng-model': 'name'}
+                ),
+                Field(
+                    'case_type',
+                    **{'ng-model': 'case_type'}
+                ),
+            ),
+            Fieldset(
+                _("Conditions"),
+                Field(
+                    'conditions',
+                    type='hidden',
+                    **{'ng-value': 'conditions'}
+                ),
+                Div(**{'ng-include': '', 'src': "'conditions.tpl'"}),
+            ),
+            Fieldset(
+               _("Action"),
+                Field(
+                    'close_case',
+                    disabled=True,
+                    checked='',
+                ),
+                Field(
+                    'update_case',
+                    **{'ng-model': 'update_case'}
+                ),
+                Div(
+                    Field(
+                        'update_property_name',
+                        **{'ng-model': 'update_property_name'}
+                    ),
+                    Field(
+                        'update_property_value',
+                        **{'ng-model': 'update_property_value'}
+                    ),
+                    **{'ng-show': 'update_case'}
+                ),
+            ),
+            FormActions(
+                StrictButton(
+                    _("Save"),
+                    type='submit',
+                    css_class='btn btn-primary col-sm-offset-1'
+                ),
+            ),
+        )
+
+    def clean_server_modified_boundary(self):
+        value = self.cleaned_data.get('server_modified_boundary')
+        if not isinstance(value, int) or value < 30:
```

Comment:
Well the reason why we made the server modified date be a required part of the criteria was for performance reasons (we don't want to be checking the rules against all the cases in the project every time).  So it seems like we need a minimum otherwise someone can just set the value to 0 and circumvent that.  I don't know if 30 is the right value, but it meets the requirements for the current projects that need this feature.